lib/src/barback/dart2js_transformer.dart

```diff
@@ -391,11 +392,20 @@ class _BarbackCompilerProvider implements dart.CompilerProvider {
     }
   }
 
-  Future<String> _readResource(Uri url) {
-    return new Future.sync(() {
+  Future _readResource(Uri url) {
+    return new Future.sync(() async {
       // Find the corresponding asset in barback.
       var id = _sourceUrlToId(url);
-      if (id != null) return _transform.readInputAsString(id);
+      if (id != null) {
+        if (id.extension == '.dill') {
+          var stream = _transform.readInput(id);
+          var bytes = <int>[];
+          await stream.forEach(bytes.addAll);
+          return bytes;
```

Comment:
Use [`collectBytes()`](https://www.dartdocs.org/documentation/async/1.13.3/async/collectBytes.html) to do this more efficiently.

Reply:
Done, thanks!